ApplicationLibCode/FileInterface/RifRoffFileTools.cpp

```diff
@@ -504,6 +506,31 @@ std::pair<bool, std::map<QString, QString>> RifRoffFileTools::createInputPropert
         return std::make_pair( false, keywordMapping );
     }
 
+    auto codeNamesAndValuesForKeyword = []( const std::string& keyword, roff::Reader& reader ) -> std::map<int, QString>
+    {
+        const std::string codeNamesKeyword  = keyword + roff::Parser::postFixCodeNames();
+        const std::string codeValuesKeyword = keyword + roff::Parser::postFixCodeValues();
+
+        auto codeNamesSize  = reader.getArrayLength( codeNamesKeyword );
+        auto codeValuesSize = reader.getArrayLength( codeValuesKeyword );
+
+        if ( codeNamesSize > 0 && codeNamesSize == codeValuesSize )
+        {
+            const auto fileCodeNames  = reader.getStringArray( codeNamesKeyword );
+            const auto fileCodeValues = reader.getIntArray( codeValuesKeyword );
+
+            std::map<int, QString> codeNamesAndValues;
+            for ( size_t i = 0; i < std::min( fileCodeValues.size(), fileCodeNames.size() ); i++ )
+            {
+                codeNamesAndValues[fileCodeValues[i]] = QString::fromStdString( fileCodeNames[i] ).trimmed();
+            }
```

Comment:
Not sure you need to use getArrayLength to get the sizes first? Just get the arrays with getStringArray/getIntArray and check if .size() matches?

std::min is also unnecessary since the sizes have to match.